app/DAO/BakeryDatabase.scala

```diff
@@ -0,0 +1,38 @@
+package DAO
+
+import javax.inject.Inject
+import play.api.db.Database
+
+import scala.concurrent.Future
+import models.DatabaseExecutionContext
+
+class BakeryDatabase @Inject() (
+    db: Database,
+    databaseExecutionContext: DatabaseExecutionContext
+) {
+
+  def getDatabaseName: Future[String] = {
+    Future {
+      db.withConnection { conn =>
+        // do something with your db
+        val result = conn.getMetaData.getTables(
+          null,
+          null,
+          null,
+          Array("TABLE")
+        )
+        var tableListStr = ""
+
+        while (result.next()) {
+          val tableName = result.getString("TABLE_NAME")
+          if (tableListStr == "") {
+            tableListStr += tableName
+          } else {
+            tableListStr = tableListStr + ", " + tableName
+          }
+        }
+        tableListStr
+      }
+    }(databaseExecutionContext)
```

Comment:
add the `implicit val` keywords on the parameter declaration and remove this

Reply:
implicit val added to the injection on line 11